appveyor_after_build.ps1

```diff
@@ -0,0 +1,300 @@
+. .\appveyor_helpers.ps1
+
+####################### Local run (Debug) configuration ###################################
+
+
+# Set this to true to be able to run outside of appveyor - just configure the section below
+$LOCAL_DEBUG_RUN = $false
+
+
+# Set these input variable that would normally come from AppVeyor yml
+if ($LOCAL_DEBUG_RUN)
+{    
+    $env:SQ_Version = "4.5.5"   
+    $env:APPVEYOR_BUILD_FOLDER = "C:\Users\bgavril\source\repos\sonar-msbuild-runner"
+    $env:configuration = "Debug"
+
+    # This script does not build the C# plugin (unlike appveyor) - please provide the path to it for the debug run
+    $localCSharpPluginPath = $env:APPVEYOR_BUILD_FOLDER + "\DeploymentArtifacts\sonar-csharp-plugin-4.2.Debug.bgavril.jar";
+}
+
+####################### Settings ###################################
+
+# URL prefix from where to download SonarQube (nb: building SQ takes too long)
+$sonarQubeDownloadUrlPrefix = "https://sonarsource.bintray.com/Distribution/sonarqube/sonarqube-";
+# Where the SonarQube zip files are downloaded (easy to cache)
+$sonarQubeDownloadPath = "c:\sonarqube\downloads";
+# Path to the SonarQube server - can be cached but then server would not be clean
+$sonarQubeInstallPath = "c:\sonarqube";
+# By default the server can be reached at localhost:9000. Using localhost. is for Fiddler to be able to capture traffic. 
+$sonarQubeBaseUrl = "http://localhost.:9000";
+# Key for the project under test
+$sqProjectKey = "ProjectUnderTest";
+# Name of the quality profile used for analysis
+$testQualityProfileName = "ProfileForTest";
+
+#
+# Download and unzip SonarSqube to a well-known directory. If the directory exists, consider it as a cache. 
+#
+function FetchAndUnzipSonarQube
+{
+    $sqVersion = $env:SQ_Version;
+    $sqUrl = $sonarQubeDownloadUrlPrefix + $sqVersion + ".zip";    
+    $sqInstallPath = [System.IO.Path]::Combine($sonarQubeInstallPath, "sonarqube-" + $sqVersion)
+    
+    if ($LOCAL_DEBUG_RUN -and [System.IO.Directory]::Exists($sqInstallPath))
+    {
+        LogAppveyorMessage -Message "SonarQube $version is already present in $sqInstallPath"
+        
+    }
+    else
+    {
+        LogAppveyorMessage -Message "Downloading SonarQube from $sqUrl"
+    
+        if ([System.String]::IsNullOrWhiteSpace($sqUrl))
+        {
+            throw "The SonarQube server download URL was not specified. Make sure the yml file has SQ_URI pointing to this URL"
+        }    
+
+        $file = [System.IO.Path]::Combine($sonarQubeDownloadPath, "sonarqube-" + $sqVersion + ".zip")               
+        FetchAndUnzip $sqUrl $sonarQubeInstallPath $file $false
+    }
+
+    return $sqInstallPath;
+}
+
+
+#
+# Adds the C# plugin to the list of Appveyor artifacts
```

Comment:
This allows us to download the "latest" C# plugin everytime the tests run. No more running after Dinesh for a snapshot!